esprima.js

```diff
@@ -1274,27 +1281,13 @@
 
                     default:
                         if (isOctalDigit(ch)) {
```

Comment:
No, not only 1 to 7 throws, all 1 to 9 throws.

https://people.mozilla.org/~jorendorff/es6-draft.html#sec-literals-string-literals

<pre>
<i>NonEscapeCharacter</i> ::
  <i>SourceCharacter</i> but not one of <i>EscapeCharacter</i> or <i>LineTerminator</i>

<i>EscapeCharacter</i> ::
  <i>SingleEscapeCharacter</i>
  <i>DecimalDigit</i>
  <b>x</b>
  <b>u</b>
</pre>